02-microarray/pathway-analysis_microarray_02_ora_with_webgestaltr.Rmd

```diff
@@ -0,0 +1,260 @@
+# Replace with a link to your desired differential expression results TSV file
+```
+
+Read in the online file that has differential expression results. 
+
+```{r}
+# Read in the contents of your differential expression results file
+dge_df <- readr::read_tsv(dge_url) # `dge_url` can be replaced with a file path to a TSV file with your desired gene list results
+```
+
+`read_tsv()` can read TSV files online and doesn't necessarily require you download the file first. 
+However, should you decide you want to download the file to your `data/` directory, you can use `download.file(dge_url, destfile = file.path(data_dir, "GSE37418_limma_results.tsv"))`.
+
+Let's take a look at what these differential expression results look like. 
+
+```{r}
+dge_df
+```
+
+Here we'll use the `SHH` groups' log fold-change > 2 and FDR < 0.05 as cutoffs for determining 
+what genes are of interest.
+
+```{r}
+upregulated_genes <- dge_df %>%
+  dplyr::filter(SHHvsOther > 2, adj.P.Val < 0.05) %>%
+  dplyr::pull(Gene)
+```
+
+Because we are testing if there is more overlap between a set of genes of interest and gene sets or pathways from some knowledge base (e.g., KEGG, Gene Ontology (GO)) than we would expect at random, we need to identify an appropriate background set. 
+Put another way, if a gene is _not measured_, it can not possibly be in our gene set of interest.
+We can provide our analysis method of choice with a reference list, which we will generate by selecting all the genes in the table of differential expression results.
+
+```{r}
+all_genes <- dge_df %>%
+  dplyr::pull(Gene)
+```
+
+## WebGestaltR
+
+We can check whether or not we need to convert to a different gene identifier by figuring out what gene identifiers `WebGestaltR` accepts for human.
+We can do this with the `listIdType()` function; the first argument is the organism name.
+
+```{r}
+listIdType("hsapiens")
+```
+
+We can see that there is `ensembl_gene_id` listed above so the Ensembl IDs in the `Gene` column are compatible identifiers, and we do not need to convert to a different identifier.
+All refine.bio dataset downloads come with Ensembl IDs but if do find yourself needing to convert gene IDs, we'd encourage you to look at our [gene ID annotation examples](https://alexslemonade.github.io/refinebio-examples/02-microarray/gene-id-annotation_microarray_01_ensembl.html).
+
+The `WebGestaltR` function is a wrapper for the [WEB-based GEne SeT AnaLysis Toolkit (WebGestalt)](http://www.webgestalt.org/) [@Wang2017].
+
+WebGestalt can perform multiple _types_ of pathway analysis. 
+Here we're using it for ORA and we will use [Gene Ontology (GO)](http://geneontology.org/docs/ontology-documentation/) biological processes as our source of gene sets.
+We can see all supported gene sets for humans with `listGeneSet()` function.
+We are using `head()` so we just get a preview and not the whole table printed out. 
```

Comment:
So one direction I could have seen this module going is a move to using `clusterProfiler` because it uses gene sets in tidy format and we could cover GSEA (pre-ranked/`fgsea`) using the same package. (We used to direct folks elsewhere for GSEA generally speaking and we likely still will for my preferred way of running GSEA...) Can you remind me what types of analyses `WebGestaltR()` can do?

Reply:
Besides ORA and GSEA, it can do NTA with either `Network Retrieval Prioritization` or `Network Expansion` methods, which I don't really understand much about it but can look into more if we are interested.

Reply:
Not particularly! What kind/type/version of GSEA does it run? Phenotype permutation?

Reply:
Their docs refer to `site weighted` GSEA or `standard` GSEA. It looks like `site weighted` you provide the gene and its log ratio of expression for phenotype vs. normal, and it looks like `standard` you provide a rank. Not sure if either of those sound like 'Phenotype permutation'.

Reply:
TBH, I'm wondering if we should even show WebGestaltR since it doesn't seem like you get a lot of functionality not already covered by their GUI (which is what our users would probably be more drawn to anyway).

Reply:
Aka I kind of like the idea of switching to clusterProfiler.

Reply:
Yea, to me, `clusterProfiler` seems to be a better fit overall.